cpp/include/cudf_test/column_wrapper.hpp

```diff
@@ -701,6 +702,340 @@ class strings_column_wrapper : public detail::column_wrapper {
   }
 };
 
+/**
+ * @brief `column_wrapper` derived class for wrapping dictionary columns.
+ *
+ * This class handles fixed-width type keys.
+ *
+ * @tparam KeyElementTo Specify a fixed-width type for the key values of the dictionary
+ * @tparam SourceElementTo For converting fixed-width values to the KeyElementTo
+ */
+template <typename KeyElementTo, typename SourceElementT = KeyElementTo>
+class dictionary_column_wrapper : public detail::column_wrapper {
+ public:
+  /**
+   * @brief Cast to dictionary_column_view
+   */
+  operator dictionary_column_view() const { return cudf::dictionary_column_view{wrapped->view()}; }
+
+  /**
+   * @brief Default constructor initializes an empty column with dictionary type.
+   */
+  dictionary_column_wrapper() : column_wrapper{}
+  {
+    wrapped = cudf::make_empty_column(cudf::data_type{cudf::type_id::DICTIONARY32});
+  }
+
+  /**
+   * @brief Construct a non-nullable dictionary column of the fixed-width elements in the
+   * range `[begin,end)`.
+   *
+   * Example:
+   * @code{.cpp}
+   * // Creates a non-nullable dictionary column of INT32 elements with 5 elements
+   * std::vector<int32_t> elements{0, 2, 2, 6, 6};
+   * dictionary_column_wrapper<int32_t> w(element.begin(), elements.end());
+   * @endcode
+   *
+   * @note Similar to `std::vector`, this "range" constructor should be used
+   *       with parentheses `()` and not braces `{}`. The latter should only
+   *       be used for the `initializer_list` constructors.
+   *
+   * @param begin The beginning of the sequence of elements
+   * @param end The end of the sequence of elements
+   */
+  template <typename InputIterator>
+  dictionary_column_wrapper(InputIterator begin, InputIterator end) : column_wrapper{}
```

Comment:
This should let you just reuse the logic already present in `fixed_width_column_wrapper`. 
```suggestion
  dictionary_column_wrapper(InputIterator begin, InputIterator end) : 
     wrapped{ encode( fixed_width_column_wrapper<KeyElementTo, SourceElementT>(begin, end) ) }
```